flink-connectors/flink-connector-pulsar/src/main/java/org/apache/flink/connector/pulsar/common/schema/factories/AvroSchemaFactory.java

```diff
@@ -45,4 +48,14 @@ public SchemaType type() {
 
         return AvroSchema.of(definition);
     }
+
+    @Override
+    public TypeInformation<T> createTypeInfo(SchemaInfo info) {
```

Comment:
For AVRO based struct, we can use Flink's built-in AVRO support as it supports schema state evolution out-of-box. There is one thing worth mentioning, we don't know what will happen if the flink AVRO class version is different from the pulsar AVRO version used to encode messages.